File: vigenere.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int j;
int main(int argc, string argv[])

{
    string k = argv[1];
    if(argc == 2)
    {
        for(int i = 0, n = strlen(k); i < n; i++)
        {
            if( (islower(k[i]) || isupper(k[i])) && isdigit(k[i]) == 0 )
            {
                j = 1;
            }
            else
            {
                j = 0;
                break;
                return 1;
            }
        }
        if(j == 1)
            {
                printf("Plaintext: ");
                string p = get_string();
                printf("ciphertext: ");
                for(int i = 0, a = 0, n = strlen(p); i < n; i++)
                {
                    int b = a % strlen(k);
                    if(isalpha(p[i]))
                    {
                        if(islower(p[i]))
                        {
                            char x = (p[i] + (tolower(k[b]) - 97) - 97) % 26;
                            printf("%c", x + 97);
                        }
                        else if(isupper(p[i]))
                        {
                            char x = (p[i] + (tolower(k[b]) - 97) - 65) % 26;
                            printf("%c", x + 65);
                        }
                        a++;
                    }
                    else
                    {
                        printf("%c", p[i]);
                    }
                }
                printf("\n");
                return 0;
            }
            else
            {
                printf("String not accepted.\n");
                return 1;
            }
    }
    else
    {
        printf("Need two arguments!\n");
        return 1;
    }
}

File: vigenere.py
import cs50
import sys

def main():

    if len(sys.argv) != 2:
        print ("usage: python vigenere.py key")
    else:
        key = sys.argv[1]

        if sys.argv[1].isalpha() == False:
            exit(1)
        else:
            print ("plaintext: ", end="")
            plain = cs50.get_string()
            print ("cyphertext: ", end="")
            a = 0

            for j in plain:
                if j.isalpha():
                    k = sys.argv[1][a]
                    if j.islower():
                        x = (((ord(j) - 97) + (ord(k.lower()) - 97)) % 26) + 97
                        print ("{}".format(chr(x)), end="")

                    elif j.isupper():
                        x = (((ord(j) - 65) + (ord(k.lower()) - 97)) % 26) + 65
                        print ("{}".format(chr(x)), end="")

                    a += 1
                    a = (a + len(key)) % len(key)

                else:
                    print ("{}".format(j), end="")

            print("\n", end="")

if __name__ == "__main__":
    main()


